src/cloud/components/Comments/ThreadItem.tsx

```diff
@@ -43,80 +42,132 @@ function ThreadItem({
   addReaction,
   removeReaction,
   user,
-}: ThreadListItemProps) {
-  const { translate } = useI18n()
+  onCommentDelete,
+}: ThreadListItemProps) => {
   const [editing, setEditing] = useState(false)
+  const { messageBox } = useDialog()
+  const { translate } = useI18n()
 
-  const [threadComments, setThreadComments] = useState<Comment[] | null>(null)
   const [showingContextMenu, setShowingContextMenu] = useState<boolean>(false)
-  const reloadComments = useCallback(async () => {
-    const comments = await listThreadComments({ id: thread.id })
-    const sortedComments = sortBy(prop('createdAt'), comments)
-    setThreadComments(sortedComments)
-  }, [thread.id])
-
-  useEffect(() => {
-    reloadComments()
-  }, [reloadComments, thread.id])
-
-  const showReplyForm = useCallback(() => {
-    if (threadComments == null || threadComments.length > 1) {
-      return
-    }
 
+  const showContextMenu = useCallback(() => {
     setShowingContextMenu(true)
-  }, [threadComments])
+  }, [])
 
-  const hideReplyForm = useCallback(() => {
+  const hideContextMenu = useCallback(() => {
     setShowingContextMenu(false)
   }, [])
 
-  const threadOpenerComment = useMemo(() => {
-    if (threadComments == null || threadComments.length == 0) {
-      return
-    }
-    return threadComments[0]
-  }, [threadComments])
-
-  const threadOpenedUser = useMemo(() => {
-    if (threadComments == null || threadComments.length == 0 || user == null) {
-      return
-    }
-
-    return threadComments[0].user
-  }, [threadComments, user])
-
   const submitComment = useCallback(
```

Comment:
the naming irks me a bit. We are not submitting a new comment
we are submitting an update to the  thread's initial/first comment. 
`updateThreadOpener` or something would be more intuitive